build_docs.pl

```diff
@@ -426,17 +457,22 @@ sub init_repos {
         say "Removing old repo <" . $dir->basename . ">";
         $dir->rmtree;
     }
-    return $repos_dir;
+    return ($repos_dir, $temp_dir, $target_repo, $target_repo_checkout);
 }
 
 #===================================
 sub push_changes {
 #===================================
-    my $build_dir = shift;
+    my ($build_dir, $target_repo, $target_repo_checkout) = @_;
 
+    local $ENV{GIT_WORK_TREE} = $target_repo_checkout if $target_repo;
```

Comment:
I'm not sure of a more clean way to manipulate the environment than this. It *works* but it doesn't look nice.